lib/src/rules/always_specify_types.dart

```diff
@@ -163,8 +204,57 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   @override
   void visitVariableDeclarationList(VariableDeclarationList list) {
-    if (list.type == null) {
-      rule.reportLintForToken(list.keyword);
+    var keyword = list.keyword;
+    if (list.type == null && keyword != null) {
+      List<String>? types;
+      var parent = list.parent;
+      if (parent is TopLevelVariableDeclaration) {
+        types = _getTypes(parent.variables);
+      } else if (parent is ForPartsWithDeclarations) {
+        types = _getTypes(parent.variables);
+      } else if (parent is FieldDeclaration) {
+        types = _getTypes(parent.fields);
+      } else if (parent is VariableDeclarationStatement) {
+        types = _getTypes(parent.variables);
+      }
+
+      if (types == null) return;
+
+      var singleType = types.toSet().length == 1;
+
+      List<Object> arguments;
+      ErrorCode? errorCode;
+      if (types.isEmpty) {
+        arguments = [];
+      } else if (keyword.type == Keyword.VAR) {
+        if (singleType) {
+          arguments = [keyword, types.first];
+          errorCode = keywordCouldBeTypeCode;
+        } else {
+          arguments = [keyword];
+          errorCode = keywordCouldBeSplitToTypesCode;
+        }
+      } else {
+        if (singleType) {
+          arguments = [types.first];
+          errorCode = specifyTypeCode;
+        } else {
+          arguments = [];
+        }
+      }
+      rule.reportLintForToken(keyword,
+          arguments: arguments, errorCode: errorCode);
+    }
+  }
+
+  List<String> _getTypes(VariableDeclarationList list) {
+    var types = <String>[];
+    for (var variable in list.variables) {
+      var type = variable.initializer?.staticType;
+      if (type != null) {
+        types.add(type.getDisplayString(withNullability: false));
```

Comment:
The fact that we're not including nullability information here means that we can't tell the difference between nullable and non-nullable types. Consider the following
```
String? s1 = '';
var s2 = '', s3 = s1;
```
I suspect that in this case we'll suggest using the type `String`, but that will lead to an error because `String?` (`s1`) isn't assignable to `String`.

Reply:
I see, fixed